test/bases/renku_data_services/data_api/test_projects.py

```diff
@@ -1314,6 +1314,94 @@ async def test_project_copy_creates_new_custom_environment_instance(
 
 
+@pytest.mark.asyncio
+async def test_project_copy_creates_new_build_and_environment_instances(
+    sanic_client,
+    user_headers,
+    regular_user,
+    create_project,
+    create_session_launcher,
+    create_project_copy,
+    create_resource_pool,
+) -> None:
+    project = await create_project("Project")
+    project_id = project["id"]
+    resource_pool = await create_resource_pool(admin=True)
+    launcher = await create_session_launcher(
+        "Launcher",
+        project_id,
+        environment={
+            "repository": "https://github.com/some/repo",
+            "builder_variant": "python",
+            "frontend_variant": "vscodium",
+            "environment_image_source": "build",
+        },
+        resource_class_id=resource_pool["classes"][0]["id"],
+        disk_storage=42,
+    )
+
+    await asyncio.sleep(1)
+
+    copy_project = await create_project_copy(project_id, regular_user.namespace.slug, "Copy Project")
+    copy_project_id = copy_project["id"]
+    _, response = await sanic_client.get(
+        f"/api/data/projects/{copy_project_id}/session_launchers", headers=user_headers
+    )
+
+    assert response.status_code == 200, response.text
+    copied_launcher = response.json[0]
+    # NOTE: Check that a new launcher is created
+    assert copied_launcher["id"] != launcher["id"]
+    assert copied_launcher["project_id"] == copy_project_id
+    assert copied_launcher["name"] == launcher["name"]
+    assert copied_launcher["description"] == launcher["description"]
+    assert copied_launcher["resource_class_id"] == launcher["resource_class_id"]
+    assert copied_launcher["disk_storage"] == launcher["disk_storage"]
+    assert copied_launcher["creation_date"] != launcher["creation_date"]
+    # NOTE: Check that a new environment is created
+    environment = copied_launcher["environment"]
+    assert environment["id"] != launcher["environment"]["id"]
+    assert environment["name"] == launcher["environment"]["name"]
+    assert environment["creation_date"] != launcher["environment"]["creation_date"]
+    assert environment["description"] == launcher["environment"]["description"]
+    assert environment["container_image"] == launcher["environment"]["container_image"]
+    assert environment["default_url"] == launcher["environment"]["default_url"]
+    assert environment["uid"] == launcher["environment"]["uid"]
+    assert environment["gid"] == launcher["environment"]["gid"]
+    assert environment.get("working_directory") == launcher["environment"].get("working_directory")
+    assert environment.get("mount_directory") == launcher["environment"].get("mount_directory")
+    assert environment["port"] == launcher["environment"]["port"]
+    assert environment.get("command") == launcher["environment"].get("command")
+    assert environment.get("args") == launcher["environment"].get("args")
+    assert environment["is_archived"] == launcher["environment"]["is_archived"]
```

Comment:
```suggestion
    assert environment["id"] != launcher["environment"]["id"]
    assert environment["creation_date"] != launcher["environment"]["creation_date"]
    assert environment == snapshot(exclude=props("id","creation_date"))
```

you could use the snapshot fixture here if you want. makes it a bit simpler. Downside is that if it returns bad values when you snapshot it, it will check against the bad values in the future. the plus side is, if we add something in the future that isn't currently covered by any of the assert, the snapshot would catch that and fail.
So maybe use it in addition to the assert if you're worried about snapshotting bad data?